applications/system-service/powerapi.cpp

```
@@ -21,29 +21,47 @@ PowerAPI::PowerAPI(QObject* parent)
+        Oxide::Sentry::sentry_span(t, "monitor", "Setup monitor", [this]{
+            if(deviceSettings.getDeviceType() == Oxide::DeviceSettings::RM1){
+                UDev::singleton()->addMonitor("platform", NULL);
+                UDev::singleton()->subsystem("power_supply", [this]{
+                    update();
+                });
+            }else{
+                timer = new QTimer(this);
+                timer->setSingleShot(false);
+                timer->setInterval(3 * 1000); // 3 seconds
+                timer->moveToThread(qApp->thread());
+                connect(timer, &QTimer::timeout, this, QOverload<>::of(&PowerAPI::update));
```

cppcheck

[error] There is an unknown macro here somewhere. Configuration is required. If Q_ENUM is a macro then please configure it.

(unknownMacro)